### PR TITLE
138 handling-dashboard-firstpage

### LINKED ISSUE #138: Handling Dashboard - FIrst Page
### Description

Create the first page of the handling dashboard with the 6 graphs shown on the mock. 

### Acceptance Criteria

- [ ] Looks like mocks
- [ ] Componentize the graph / use the already existing graph component
- [ ] Communicates with backend to query the necessary data either through the socket or http requests

### Proposed Solution

^

### Mocks

epic #137 
Screenshot 2024-03-02 at 8 31 43 PM

### PR DESCRIPTION
**Pull request by leoleader**
added mock data af

still TO DO:
- update graph component to add multiple line functionality
- radar chart

## Changes

- set up handling page routing and in sidebar 
- created handling page component
- created 6 specific graph components 
- made paginator for handling page

## Notes

one of the graphs is a line chart with multiple lines and rn our graph / info-graph components are hardcoded to just do single line. i made a comment while coding about a proposed solution which i pasted below:

  // currently info-graph and graph component hardcoded to do only single line, line charts.
  // instead of having GraphData as x,y then setting series = {name: blah, data = data} inside graphcomponent
  // i think have option for incoming graph component data be an additional type called SeriesData that is series form
  // {name: blah, data = data}[] and then act accordingly based off given type.

## Screenshots
adding mock data 
Screenshot 2024-05-10 at 2 52 36 PM

Screenshot 2024-04-17 at 8 05 07 PM

older screenshot showing side bar

Screenshot 2024-04-17 at 6 52 16 PM

## To Do

- test graphs actually work as expected with data

Closes #138

### CODE REVIEW

File: angular-client/src/components/handling-graphs/gas-brake-pos/gas-brake-pos.component.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  templateUrl: './gas-brake-pos.component.html',
+  styleUrls: ['./gas-brake-pos.component.css']
+})
+export default class GasBrakePos implements OnInit {
+  gasData: GraphData[] = [];
+  brakeData: GraphData[] = [];
+  // currently info-graph and graph component hardcoded to do only single line, line charts.
+  // instead of having GraphData as x,y then setting series = {name: blah, data = data} inside graphcomponent
+  // i think have option for incoming graph component data be an additional type called SeriesData that is series form
+  // {name: blah, data = data}[] and then act accordingly based off given type.
```
Comment (Peyton-McKee) [inline angular-client/src/components/handling-graphs/gas-brake-pos/gas-brake-pos.component.ts:19]: yeah that would mean you could graph multiple lines on a graph

File: angular-client/src/components/handling-graphs/long-vs-lat/long-vs-lat.component.html [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<info-graph [data]="data" graphContainerId="longVsLatGraph" title="Longitude vs Latitude Acceleration" color="primary">
```
Comment (Peyton-McKee) [inline angular-client/src/components/handling-graphs/long-vs-lat/long-vs-lat.component.html:1]: longvslat wont work as a line graph, https://apexcharts.com/javascript-chart-demos/radar-charts/ try using this kind of graph

File: angular-client/src/components/handling-graphs/steerPos-vs-time/steerPos-vs-time.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  styleUrls: ['./steerPos-vs-time.component.css']
+})
+export default class SteerPosVsTime implements OnInit {
+  data: GraphData[] = [];
+  maxDataPoints = 100;
+
+  constructor(private storage: Storage) {}
+  ngOnInit() {
+    this.storage.get(IdentifierDataType.STEERING_ANGLE).subscribe((value) => {
+      this.data.push({ x: new Date().getTime(), y: parseInt(value.values[0]) });
```
Comment (Peyton-McKee) [inline angular-client/src/components/handling-graphs/steerPos-vs-time/steerPos-vs-time.component.ts]: use the tiemstamp from the value,

File: angular-client/src/components/handling-graphs/shot-vs-time/shot-vs-time.component.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  styleUrls: ['./shot-vs-time.component.css']
+})
+export default class ShotVsTime implements OnInit {
+  data: GraphData[] = [];
+  maxDataPoints = 100;
+
+  constructor(private storage: Storage) {}
+  ngOnInit() {
+    this.storage.get(IdentifierDataType.SHOT).subscribe((value) => {
+      this.data.push({ x: new Date().getTime(), y: parseInt(value.values[0]) });
```
Comment (Peyton-McKee) [inline angular-client/src/components/handling-graphs/shot-vs-time/shot-vs-time.component.ts]: use timestamp from the value

File: angular-client/src/components/handling-graphs/ride-height-time/ride-height-time.component.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  styleUrls: ['./ride-height-time.component.css']
+})
+export default class RideHeightTime implements OnInit {
+  data: GraphData[] = [];
+  maxDataPoints = 100;
+
+  constructor(private storage: Storage) {}
+  ngOnInit() {
+    this.storage.get(IdentifierDataType.RIDEHEIGHT).subscribe((value) => {
+      this.data.push({ x: new Date().getTime(), y: parseInt(value.values[0]) });
```
Comment (Peyton-McKee) [inline angular-client/src/components/handling-graphs/ride-height-time/ride-height-time.component.ts]: use timestamp from the value

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
This looks great. Can you just add some mock data to the backend to test it as well

**Comment by jr1221** [comment]
Brake pedal topic: `Pedals-Brake_1` (range 0 to ~800 ish)
Accel pedal topoc: `Pedals-Accelerator_1` (range ~1000 to ~4500 ish)

There is `_2` for both, but idk if you need to read both sensors.

**Comment by RChandler234** [comment]
Stale asf

**Pull request closed without merging**